### ISSUE TITLE
Read-Only filesystem on Ubiquiti airmax devices

### ISSUE THREAD

**Issue by blocktrron** [freifunk-gluon/gluon#2939]
### Bug report

**What is the problem?**

Several Ubiquiti airmax devices report a Read-only filesystem with OpenWrt 23.05

Upstream ticket: https://github.com/openwrt/openwrt/issues/12882

**What is the expected behaviour?**

The rootfs_data partition is not read-only

**Gluon Version:**

Gluon based on OpenWrt 23.05

**Site Configuration:**
n/a

**Custom patches:**
n/a

**Comment by blocktrron** [comment]
Removing this ticket for the upcoming v2023.2 milestone, as the devices were removed from Gluon. 


Keeping this ticket open for visibility.

**Comment by rotanid** [comment]
@blocktrron this seems to be solved upstream?

**Comment by blocktrron** [comment]
DO you have a commit as reference?

**Comment by rotanid** [comment]
@blocktrron f024f4b1b0380b3b2e18115bd8e4f35393fccc70 was mentioned here https://github.com/openwrt/openwrt/pull/14237

**Comment by herbetom** [comment]
I could validate te fix on the NanoBeam 5AC 19 (XC). Hoewer that device is currently in Darmstadt and it's currently raining which doesn't help with my motivation to do so right now.

Hoewer, in theory we could validate it and reintroduce support before the next release.

**Comment by blocktrron** [comment]
@rotanid I've seen this commit, I'd like to have this verified before reverting though.

**Comment by nrbffs** [comment]
We have M5 Loco XM (5x) and M5 Loco XW (3x). Is the fix already in master? What needs to be done to verify this?

**Comment by rotanid** [comment]
https://github.com/openwrt/openwrt/commit/c55aaa7c9a98e7c0a5e1ea8293a534dc5b395cf3

@blocktrron shouldn't this be part of gluon already which would render this issue fixed?

**Comment by nrbffs** [comment]
As far as I understand, the fix from OpenWRT commit c55aaa7c9a ("ath79: generic: disable SPI-NOR write protect unconditionally") is now in Gluon main. 

To drive this issue forward, I would do the following:

- make a build of latest gluon main, with reverted a0f5b4e9468 ("ath79-generic: remove Ubiquiti AirMax targets (#3067)")
- flash it onto a Ubiquiti Nanostation Loco M (XW) or Ubiquiti Nanostation M (XW)  - does not matter if M2 or M5
- using `cat /proc/mtd`, find the mtd device which corresponds to the firmware partition
-  `cat /sys/block/mtdblock*/ro` and verify that it shows `0`
- click through configmode and verify the device reboots fine and persists its config (i.e. doesn't come up in config mode again)
- sysupgrade (loosing settings) to an older version to verify upgrade works

If the above test was successful and nobody objects until next week, I will prepare a PR for revert of a0f5b4e9468.

**Comment by herbetom** [comment]
@nrbffs Yes, let's get this hopefully over with. I've rebased #3141 which is a revert of a0f5b4e9468e9c2fa28459143a7df6ced348f452. Locally it seems to build. So i think the CI should succeed as well.

The testing procedure sounds fine to me.

I'll test it on the NanoBeam 5AC 19 (XC) today or tommorrow and if you can also provide some testing results (ideally from another affected Model) then we should have something mergeable with a reasonable high degree of confidence.

**Comment by herbetom** [comment]
These appear to be the current Upstream Issues:
- https://github.com/openwrt/openwrt/issues/17234
- https://github.com/openwrt/openwrt/issues/17285

(But there are more.)

And then there is a PR which claims to fix it:
- https://github.com/openwrt/openwrt/pull/17287

**Comment by FGBxRamel** [comment]
As the mentioned issues are closed and the last comment on them and the PR mention that fixes are in the Linux kernel, I'll try to test if this problem still exists. Hopefully I'll find fitting devices, maybe someone else can test again too.
